src/wgsl_parser.ts

```diff
@@ -627,9 +627,9 @@ export class WgslParser {
     const condition = this._optional_paren_expression();
     const block = this._compound_statement();
 
-    let elseif: Array<AST.ElseIf> | null = null;
-    if (this._match(TokenTypes.keywords.elseif))
-      elseif = this._elseif_statement();
+    let elseif: Array<AST.ElseIf> | null = [];
+    if (this._match_elseif())
+    { elseif = this._elseif_statement(elseif); }

```

Comment:
If you're going to use braces here, please match the style of the existing code.